pkg/controller/controlplane/valuesprovider.go

```diff
@@ -461,18 +466,19 @@ func (vp *valuesProvider) getCCMChartValues(
 			"server": serverSecret.Name,
 		},
 		"gep19Monitoring": gep19Monitoring,
+		"configureCloudRoutes": func() bool {
+			if isDualstackEnabled(cluster.Shoot.Spec.Networking) {
+				return false
+			}
+
+			return !overlayEnabled
+		}(),
 	}
 
 	if cpConfig.CloudControllerManager != nil {
 		values["featureGates"] = cpConfig.CloudControllerManager.FeatureGates
 	}
 
-	overlayEnabled, err := vp.isOverlayEnabled(cluster.Shoot.Spec.Networking)
-	if err != nil {
-		return nil, err
-	}
-	values["configureCloudRoutes"] = !overlayEnabled || !isDualstackEnabled(cluster.Shoot.Spec.Networking)
```

Comment:
I think `values["configureCloudRoutes"] = !overlayEnabled && !isDualstackEnabled(cluster.Shoot.Spec.Networking)` would be correct.